Remotion/Data/DomainObjects/Mapping/ClassDefinition.cs

```diff
@@ -120,61 +90,47 @@ public bool IsSameOrBaseClassOf (ClassDefinition classDefinition)
       return false;
     }
 
-    public ClassDefinition[] GetAllDerivedClasses ()
-    {
-      var allDerivedClasses = new List<ClassDefinition>();
-      FillAllDerivedClasses(allDerivedClasses);
-      return allDerivedClasses.ToArray();
-    }
-
-    public ClassDefinition GetInheritanceRootClass ()
+    public override bool IsAssignableFrom (TypeDefinition other)
     {
-      if (BaseClass != null)
-        return BaseClass.GetInheritanceRootClass();
+      if (ReferenceEquals(this, other))
```

Comment:
SHould we do this as a basecall instead? the base implementation also does referencequals and we ignore the value implementation here completely. Typically, there should be a reason the base call is omitted and I don't see it here. Thoughts?

Reply:
I think you mixed up `TypeDefinition` and `TypeDefinitionForUnresolvedRelationPropertyType` as the method is abstract and thus not implemented in `TypeDefinition`. I also would not make it virtual as this logic should always be implemented by a subclass.